src/core/useMutableSource.ts

```diff
@@ -42,13 +42,15 @@ export const useMutableSource = (
     (currentVersion !== state[3] && currentVersion !== lastVersion.current)
   ) {
     currentSnapshot = getSnapshot(source[TARGET])
-    setState([
-      /* [0] */ source,
-      /* [1] */ getSnapshot,
-      /* [2] */ subscribe,
-      /* [3] */ currentVersion,
-      /* [4] */ currentSnapshot,
-    ])
+    if (currentSnapshot !== state[4]) {
```

Comment:
I couldn't find the right solution that stays outside of this file, but this let the tests pass.

Reply:
Thanks for investigating. I know what this means because I did this once.
At that time I found another solution in vanilla.ts. Maybe, this one is not avoidable.
One thing for sure is we might not be using the "version" correctly.
In any case, we don't know the final behavior of uMS, but pretty sure it doesn't rely on setState bail out like this emulation.
So, the expectation is it will be fixed in the future.
Let me think if this fix makes sense for other cases.

Reply:
Yea, ok, if you found another solution, push it, no problem. it's a draft for now.